src/pool-bin/libraries/BinPool.sol

```diff
@@ -464,12 +474,24 @@ library BinPool {
     function _subShare(State storage self, address owner, uint24 binId, bytes32 salt, uint256 shares) internal {
         self.positions.get(owner, binId, salt).subShare(shares);
         self.shareOfBin[binId] -= shares;
+
+        /// @dev Ensure bin user's position share is either 0 or greater than minimum share
+        uint256 userPositionShare = self.positions.get(owner, binId, salt).share;
+        if (userPositionShare > 0 && userPositionShare < MINIMUM_SHARE) {
```

Comment:
```suggestion
        if (userPositionShare < MINIMUM_SHARE) {
```
I think this will be sufficient

Reply:
no , if removed all , userPositionShare will be 0 now , will get revert error